### PR TITLE
 Closes #324: Add support for solr in integration tests 

### LINKED ISSUE #324: Extend integration tests to use solr dedicated for tests

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java [hunk 1]
```diff
@@ -86,6 +99,45 @@ public String getSetting(String key) {
...
+            throw new RuntimeException("Unable to read properties from classpath: " + classpath, e);
+        }
+        return properties;
+    }
+    
+    private Properties loadFilesystemProperties(String path, boolean optional) {
+        Properties properties = new Properties();
+        File propertiesFile = new File(path);
+
+        if (!propertiesFile.exists() || !propertiesFile.isFile()) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java]: Isn't `!(propertiesFile.exists() && propertiesFile.isFile())` a little clearer?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java [hunk 2]
```diff
@@ -86,6 +99,45 @@ public String getSetting(String key) {
 
     // -------------------- PRIVATE --------------------
 
+    private Properties loadClasspathProperties(String classpath, boolean optional) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java]: Can we refactor `optional` to `isOptional` ?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java [hunk 3]
```diff
@@ -86,6 +99,45 @@ public String getSetting(String key) {
...
+            }
+            
+            properties.load(propertiesInputStream);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to read properties from classpath: " + classpath, e);
+        }
+        return properties;
+    }
+    
+    private Properties loadFilesystemProperties(String path, boolean optional) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java]: Can we refactor `optional` to `isOptional` ?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    
+    // -------------------- GETTERS --------------------
+    
+    public List<SettingLocation> getSettingLocations() {
+        return Collections.unmodifiableList(settingLocations);
+    }
+
+    // -------------------- LOGIC --------------------
+    
+    public void addLocation(SettingLocationType locationType, String path, boolean optional) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: Can we refactor `optional` to `isOptional` ?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        settingLocations.add(new SettingLocation(locationType, path, optional));
+    }
+
+
+    // -------------------- INNER CLASSES --------------------
+    
+    public static class SettingLocation {
+        private SettingLocationType locationType;
+        private String path;
+        private boolean optional;
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: Can we refactor `optional` to `isOptional` ?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        
+        // -------------------- CONSTRUCTORS --------------------
+        
+        private SettingLocation(SettingLocationType locationType, String path, boolean optional) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: Can we refactor `optional` to `isOptional` ?

File: dataverse-webapp/src/main/resources/log4j.properties [hunk 7]
```diff
@@ -1,4 +1,4 @@
-log4j.rootLogger=FATAL, stderr
+log4j.rootLogger=INFO, stdout
```
Comment (dbojanek) [inline dataverse-webapp/src/main/resources/log4j.properties:1]: Do we want to keep this?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/ArquillianExampleIT.java [hunk 8]
```diff
@@ -2,16 +2,34 @@
...
+import org.testcontainers.utility.MountableFile;
 
 import javax.ejb.EJB;
+import javax.inject.Inject;
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
+
+import java.io.IOException;
 import java.util.List;
+import java.util.logging.Logger;
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/ArquillianExampleIT.java]: Some imports are unused.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    public void loadSettings__NOT_EXISTING_PROPERTIES_FROM_CLASSPATH() {
+        
+        // given
+        FileSettingLocations settingLocations = new FileSettingLocations();
+        settingLocations.addLocation(SettingLocationType.CLASSPATH, "/notexisting.properties", false);
+        
+        FileBasedSettingsFetcher settingsFetcher = new FileBasedSettingsFetcher(settingLocations);
+        
+        // when
+        Executable loadSettingsOperation = () -> settingsFetcher.loadSettings();
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java]: Isn't `settingsFetcher::loadSettings` more readable?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+    public void loadSettings__NOT_EXISTING_PROPERTIES_FROM_FILESYSTEM() {
+        
+        // given
+        FileSettingLocations settingLocations = new FileSettingLocations();
+        settingLocations.addLocation(SettingLocationType.FILESYSTEM, "notexisting.properties", false);
+        
+        FileBasedSettingsFetcher settingsFetcher = new FileBasedSettingsFetcher(settingLocations);
+        
+        // when
+        Executable loadSettingsOperation = () -> settingsFetcher.loadSettings();
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java]: Isn't `settingsFetcher::loadSettings` more readable?

**Review by dbojanek — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java (hunk 1 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java]: I will change it

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java (hunk 2 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java (hunk 3 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcher.java]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java (hunk 4 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java (hunk 5 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: And change getter to getIsOptional()?

Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java (hunk 6 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/settings/FileSettingLocations.java]: ok

File: dataverse-webapp/src/main/resources/log4j.properties (hunk 7 above)
Comment (madryk) [reply dataverse-webapp/src/main/resources/log4j.properties:1]: It depends. Without this we wouldn't have any logs from solr container

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/ArquillianExampleIT.java (hunk 8 above)
Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/ArquillianExampleIT.java]: I will organize imports

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java (hunk 9 above)
Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java]: Ok. I will change it

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java (hunk 10 above)
Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java]: Ok. I will change it

**Review by dbojanek — APPROVED** [review]

**Pull request merged**